packages/ccda/src/ccda-to-fhir.ts

```diff
@@ -136,11 +213,1056 @@ export function convertCcdaToFhir(ccda: Ccda, options?: CcdaToFhirOptions): Bund
+        id: { extension: '1250504853', root: OID_US_NPI_CODE_SYSTEM },
+        addr: {
+          streetAddressLine: '123 Happy St',
+          city: 'Sunnyvale',
+          state: 'CA',
+          postalCode: '95008',
+          country: 'US',
+        },
+        telecom: { use: 'WP', value: 'tel:(781)271-3000' },
+        assignedAuthoringDevice: {
+          manufacturerModelName: 'Medplum Test System',
+          softwareName: 'Medplum Test System',
+        },
+      },
+    },
+    custodian: {
+      assignedCustodian: {
+        representedCustodianOrganization: {
+          id: { extension: '117323', root: '2.16.840.1.113883.4.336' },
+          name: 'Medplum Test Deck',
+          telecom: { use: 'WP', value: 'tel:(781)271-3000' },
+          addr: {
+            use: 'HP',
+            streetAddressLine: '202 Burlington Rd.',
+            city: 'Bedford',
+            state: 'MA',
+            postalCode: '01730',
+            country: 'US',
+          },
+        },
+      },
+    },
+    legalAuthenticator: {
+      time: currentDateTime,
+      signatureCode: { code: 'S' },
+      assignedEntity: {
+        id: { root: randomUUID() },
+        addr: {
+          streetAddressLine: '123 Happy St',
+          city: 'Sunnyvale',
+          state: 'CA',
+          postalCode: '95008',
+          country: 'US',
+        },
+        telecom: { use: 'WP', value: 'tel:(781)271-3000' },
+        assignedPerson: {
+          name: {
+            given: 'John',
+            family: 'Doe',
+          },
+        },
+        representedOrganization: {
+          id: { root: '2.16.840.1.113883.19.5' },
+          name: 'Medplum Test System',
+        },
+      },
+    },
+  };
+}
+
+/**
+ * Shared XML building utilities for both C-CDA and QRDA generation
+ */
+
+/**
+ * Creates a standardized XML builder configuration for both C-CDA and QRDA
+ * @returns XMLBuilder configuration object
+ */
+function createXmlBuilderConfig(): any {
+  return {
+    ignoreAttributes: false,
+    attributeNamePrefix: '@_',
+    format: true,
+    indentBy: '  ',
+    suppressBooleanAttributes: false,
+  };
+}
+
+/**
+ * Builds XML header for both C-CDA and QRDA documents
+ * @returns XML header string
+ */
+function buildXmlHeader(): string {
+  return '<?xml version="1.0" encoding="utf-8"?>\n';
+}
+
+/**
+ * Converts a data object to XML string using standardized configuration
+ * @param data - The data object to convert
+ * @returns XML string
+ */
+function buildXmlString(data: Record<string, any>): string {
+  const builder = new XMLBuilder(createXmlBuilderConfig());
+  return buildXmlHeader() + builder.build(data);
+}
+
+/**
+ * Shared code mapping utilities for both C-CDA and QRDA generation
+ */
+
+/**
+ * Maps a FHIR code to a standardized format for both C-CDA and QRDA
+ * @param code - The FHIR CodeableConcept
+ * @param defaultSystem - Default code system if not specified
+ * @returns Standardized code object
+ */
+function mapCodeToStandard(
```

Comment:
This function can be removed.  It is a complete duplicate of `mapCodeableConceptToCcdaCode` in `systems.ts`